### ISSUE TITLE
Inconvenient auto scroll when editing note

### ISSUE THREAD

**Issue by GuyOlivier** [laurent22/joplin#1546]
### Operating system
- Windows

### Application
- Desktop

First thanx for this great software.

I use a lot of code in my notes. I noticed that when I use piece of code
```
of this kind
```
or when I insert images
and when my notes are longer than one screen, the editing screen regularly scrolls to the top (step by  step). But finally it scrolls in such a way that the line with the cursor moves above the top of the screen. So the line where the cursor is currently focused disappears above the top and will reappear when I start typing again. After a few more typing it will scroll again above the top and so forth.
In appearance parameter I did not select any Editor font family.

**Comment by tessus** [comment]
There is no auto-scroll in Joplin.

**Comment by GuyOlivier** [comment]
I record a video to demonstrate the issue I meet.
[showmore.com](https://showmore.com/u/uziald5)
You can skip the first 45 seconds to see directly the issue.

**Comment by i-need-to-tell-you-something** [comment]
> There is no auto-scroll in Joplin.

It is this feature where Note editor and Note viewer try to match up where they have scrolled. There is some kind of issue there.

It happened to me as well although I didn't study it this carefully. I just felt it was jumping around when there were codeblocks involved.

**Comment by tessus** [comment]
@GuyOlivier Thanks for the video. I tried the same thing, but still couldn't reproduce it.

Let's see if @laurent22 has an idea what is happening here.

**Comment by i-need-to-tell-you-something** [comment]
> @GuyOlivier Thanks for the video. I tried the same thing, but still couldn't reproduce it.
> 
> Let's see if @laurent22 has an idea what is happening here.

Did you also try it at the upper edge of the displayed text area?

**Comment by GuyOlivier** [comment]
Hello, thanx for taking a look at my issue.

@i-need-to-tell-you-something I do not understand what you mean by:
> Did you also try it at the upper edge of the displayed text area?

I am already at the top edge of the displayed area.

Here is an easy  step by step to reproduce the issue.
* Get my Joplin note at [github.com/GuyOlivier](https://github.com/GuyOlivier/Joplin/blob/master/Procedure/Upgrade%20DB%20PYX%20schema.md), switch to edit mode and copy the markdown
* Create a new note in Joplin and paste the markdown
* Scroll the text to "line 4" at the top of the screen
* Start typing text, the issue is reproduced.

I recorded a video of these steps to reproduce the error: [reproduce issue video](https://showmore.com/u/eh2m7ir)

You wonder why I am always editing the line at the top of the screen (I could move my editing line in the middle of the screen). It is because step by step the text move to the top. So when typing a quite long piece of text I will end up editing the line at the top of the screen.

**Comment by i-need-to-tell-you-something** [comment]
> @i-need-to-tell-you-something I do not understand what you mean by:

@GuyOlivier Sorry I meant that as a question to tessus. I thought quotation would be enough for that and I forgot to use the @ tag.

**Comment by tessus** [comment]
@i-need-to-tell-you-something 

> Did you also try it at the upper edge of the displayed text area?

Yes, but I can still not reproduce the problem. As mentioned before, I think @laurent22 might be able to shed some light on this.

**Comment by alexdevero** [comment]
I am experiencing the same issue. In a file with enough content to show scrollbar, when I add or remove a line, editor will automatically scroll up or down a little bit.

![joplin](https://user-images.githubusercontent.com/4237724/59156187-60c92300-8a97-11e9-892e-0e3f08b36714.gif)

**Comment by tessus** [comment]
Ok, I finally found a note where I could reproduce this issue.

I believe it has something to do with how Joplin saves the cursor position, but I'm not sure.
@laurent22, shall I export the note and attach it here, or have you seen this behavior before?

**Comment by tessus** [comment]
@laurent22 this is happening more and more often now and it makes editing a note almost impossible. The only way to edit a note, which exhibits this behavior, is to use an external editor.

**Comment by laurent22** [comment]
There was a recent PR which, among other things, changed something to scrolling, in relation to link anchors but maybe it introduced this bug too.

**Comment by laurent22** [comment]
If you can share a note where it always happen it might help

**Comment by tessus** [comment]
@laurent22 here you go: [git.jex.txt](https://github.com/laurent22/joplin/files/3384462/git.jex.txt) (remove the .txt extension....)

**Comment by tessus** [comment]
Oh, and the OP also shared his note: https://github.com/GuyOlivier/Joplin/blob/master/Procedure/Upgrade%20DB%20PYX%20schema.md

**Comment by doc75** [comment]
I confirm the issue is still present in v1.0.165.
This is making really difficult to write long notes as at some point we cannot see anymore the test we are editing.
Sorry, but I am not familiar enough with electron app to help on this.

**Comment by carlosedp** [comment]
I'm seeing this problem as well while editing notes with code blocks. Practically impossible to use with this behavior.

**Comment by tessus** [comment]
@laurent22 have you been able to reproduce this issue? it's really hard to edit a note when this happens. As mentioned before, I believe it has to do with how Joplin saves the cursor position. But maybe I'm wrong.

**Comment by mic704b** [comment]
+1  sometimes while I am editing, the line being edited jumps around erratically as I type and then moves completely off the screen!

It would be ok if the display pane moves around (a little), but I think it would be nicer if the editing pane did not.

**Comment by mgrandi** [comment]
I have several notes that exhibit this behavior, but it seems that notes that are either very long or that have lots of code blocks make it worse

**Comment by ioogithub** [comment]
Yes, I too can confirm this behavior is still happening on 1.0.174 and it drives me crazy. Joplin keeps scrolling with every new line until it has scrolled right off the screen. It really disrupts the workflow, I have actually started using a regular text editor outside of Joplin and pasting the work into Joplin in blocks. This way I can still get the benefit of backing up and securing the data but it means that basically Joplin is unusable for taking these notes and this is far from an ideal workaround.

**Comment by i-need-to-tell-you-something** [comment]
@ioogithub Instead of copy-pasting the text, can't you just use the external editor option? You can point to the path of the executable you wish to use in <kbd>Options</kbd> > <kbd>General</kbd> > <kbd>Text Editor Command</kbd>. The default hotkey to use it then when a note is selected is <kbd>CTRL</kbd>+<kbd>E</kbd>.

**Comment by Vitorvlv** [comment]
Hey, just wanted to say that this issue is still alive and well.

Here is a short video showing it: https://streamable.com/gzd82#

I'm on Joplin 1.0.179. This is a very long note, is has chemical notation in it and a few images. Restarting Joplin seems to make the issue go away, at least for the moment.

**Comment by Nowasky** [comment]
I'm experiencing the same issue on my Windows machine, adding the fact that the editor hangs for a second each time the preview screen is updated.

**Comment by scottwilliambeasley** [comment]
I can confirm that I am experiencing this same issue.

Joplin Desktop Version: 1.0.179
OS: MacOS (Catalina) 10.15.2

**Comment by Liuhaai** [comment]
Hey, @tessus . I am debugging this issue. This issue might be caused by scroll events listener.
If several lines are written into a long text, the onScroll event will be triggered. Then the editor window will be reset according to the current scroll position. However, editor window position isn't calculated accurately because of the precision of float in Javascript. Idealy, the longer the text is, the severer the problem is. I try to comment out the last line of these codes, which is to reset scroll position, and this issue is solved, and no new related bug is found currently. I have no idea what scenario is these codes aim at. At the most of the time when scrolling, the function will return directly(return; in line 4).

![image](https://user-images.githubusercontent.com/55118568/75650675-7c6aef80-5c0b-11ea-8833-dbbe71c88f64.png)

**Comment by tessus** [comment]
@Liuhaai I always had the feeling that it had something to do with cursor position handler somehow. I'm not familiar with this part of the code at all. Maybe @mic704b or @laurent22 can help you out here.

**Comment by Liuhaai** [comment]
Sorry for the late feedback. The scroll event might be triggered by some text input in the editor. I fixed this issue by a workaround. The editor position will not change when the user edit in the editor. That is, there are some changes on the content length.

**Comment by tessus** [comment]
@Liuhaai if your change fixes this problem, please submit a PR.

**Comment by Yobel16** [comment]
I have the same issue. Joplin keeps on scrolling up whenever I stop typing in the editor and this only happens when I include the mermaid syntax in one of my notes.

**Comment by akittas** [comment]
> I have the same issue. Joplin keeps on scrolling up whenever I stop typing in the editor and this only happens when I include the mermaid syntax in one of my notes.

I can confirm the same thing. After disabling mermaid syntax, Joplin works as it should, otherwise this erratic autoscroll makes it so hard to edit notes

**Comment by Yobel16** [comment]
> > I have the same issue. Joplin keeps on scrolling up whenever I stop typing in the editor and this only happens when I include the mermaid syntax in one of my notes.
> 
> I can confirm the same thing. After disabling mermaid syntax, Joplin works as it should, otherwise this erratic autoscroll makes it so hard to edit notes

Is there a fix on this? Being able to use mermaid syntax and having no problem.

**Comment by 5inful1** [comment]
I don't think it has anything to do with mermaid syntax. Nor does it require alot of text to create this issue.

If I reduce the size of my window, insert a line space and proceed to add content above this line space. The editing page seems to update as the content on the right plane updates.

![Screen-Recording-2020-05-20-at-2](https://user-images.githubusercontent.com/64338212/82352216-ff4c5b00-9a40-11ea-85c5-7ff7ce4388a5.gif)

I'm not a programmer but it feels like the view is adjusting relative to be bottom of content which is subject to change as the user types stuff rather than the content at the top of the viewer.  I hope this helps.

**Comment by koelschlaeger** [comment]
I can confirm in version:
> Joplin 1.0.201 (prod, win32)
> Sync Version: 1
> Profile Version: 28
> Revision: e65af8c1 (master)

that the behavior exists and I was able to reproduce it with these steps:
1. created a massive wall of Lorem Ipsum (no issues)
2. added basic MD objects, e.g. bulleted list, checkboxes, headings, etc (no issues)
3. added some KaTeX for giggles (no issues)
4. added a mermaid class diagram (now I have the page scrolling issue)

For every new character inserted an any block the rendered page (an therefore the raw MD) scrolls up by at least a line.

TL;DR at least in my case it is isolated to after I add the first instance of a mermaid block.

**Comment by 5inful1** [comment]
OMG Thankyou!! I think this issue has been fixed! 

Device: Mac
Version: Joplin 1.0.216 ( although the fix may have been implemented before this version)

I'm so happy, can anyone else confirm this.

I just tried editing in my one of my longest notes filled with mermaid syntax and javascript syntax the editor doesn't budge awesome. (unless I start typing at the bottom of the file which would be expected).

Thank you again to whomever fixed this.

**Comment by AlanCoding** [comment]
I'm on 1.1.4 on Fedora, and I'm still seeing this issue. I had the same scrolling issues on a Mac before I migrated to this computer.

If I close the application and then re-open it, the issue goes away... but it will eventually come back. I'm not sure what the exact thing I did that causes it is.

**Comment by aReaSa** [comment]
I, too, confirming the issue. 
Joplin for Desktop
Joplin 1.7.11
OS: Lubuntu

It scrolls to upward direction. When I continue to the typing it comes the back line I work, however, it is constantly trying to go upside. If there is solution, please do inform me. Thank you.

**Comment by KhaaL** [comment]
I'm using Joplin 2.6.6 and am experiencing this issue. I do have mermaid codeblocks in my notes, and disabling them stops this behaviour. However, that isn't a solution for me as I need mermaid in my notes...

**Comment by taw00** [comment]
This issue never seems to be resolved. Can we just please have the option to have the views not linked? I have a lot of markdown blended with a bunch of HTML, and they just become impossible to edit.

Sometimes it calms down if you browse away (click to the last note and then come back) but ... often it doesn't not.

If I could we could just disconnect the scrolling, between the two; that would solve so many problems.

**Comment by laurent22** [comment]
@taw00, do you have an example note that can be used to consistently replicate the issue? And if so, please create a new issue with this information so that hopefully it can be investigated.

**Comment by taw00** [comment]
_Note: I only use the plain-text editor._

@laurent22 I will put together a hopefully succinct example that consistently causes me issues. I use Joplin for general note-taking and things. But also often use a bit of CSS and HTML  to tweak things here and there. But the most egregious offenders are when I use Joplin to develop anything and everything that involves markdown, including markdown-formatted web pages which are then plugged into Ghost (a web CMS like Wordpress). It just seems like it would be relatively easy (I am probably being naive) to have the cursor simply not move if you are in the editor. Simply viewing and scrolling the editor text, or the rendered view is not at issue. That is fine. Everything goes south when you start to edit your document in not only the two-pane view, but the editor-only view as well. Everything scrolls okay until you try you start typing. And then it starts to jump around.

Anyway. I will put together something consistently annoying and submit a new issue.
